src/Domain/Python/FunctionsOfTime.cpp

```diff
@@ -21,22 +23,99 @@ namespace domain::py_bindings {
 
 void bind_functions_of_time(py::module& m) {  // NOLINT
   domain::FunctionsOfTime::register_derived_with_charm();
-  py::class_<FunctionsOfTime::FunctionOfTime>(m, "FunctionOfTime")
+  py::class_<FunctionsOfTime::FunctionOfTime,
+             std::shared_ptr<FunctionsOfTime::FunctionOfTime>>(m,
+                                                               "FunctionOfTime")
       .def("time_bounds", &FunctionsOfTime::FunctionOfTime::time_bounds)
       .def("func", &FunctionsOfTime::FunctionOfTime::func, py::arg("t"))
       .def("func_and_deriv", &FunctionsOfTime::FunctionOfTime::func_and_deriv,
            py::arg("t"))
       .def("func_and_2_derivs",
            &FunctionsOfTime::FunctionOfTime::func_and_2_derivs, py::arg("t"));
+  m.def(
+      "serialize_functions_of_time",
+      [](const std::unordered_map<
+          std::string,
+          std::shared_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+             shared_deserialized_functions_of_time) {
```

Comment:
You can just call this argument `functions_of_time`. The fact that it's shared is just a python thing, and they are clearly deserialized (since it's not a `vector<char>`)